File: market/test/java/ru/yandex/market/pers/pay/controller/PageMatcherControllerTest.java
package ru.yandex.market.pers.pay.controller;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

import ru.yandex.market.pers.pay.PersPayTest;
import ru.yandex.market.pers.pay.mvc.SystemMvcMocks;

import static java.lang.String.join;
import static org.junit.jupiter.api.Assertions.assertTrue;

/**
 * @author Ilya Kislitsyn / ilyakis@ / 11.03.2021
 */
public class PageMatcherControllerTest extends PersPayTest {

    private static final String SEPARATOR = "\t";
    private static final String PAGE_TYPE = "http";

    @Autowired
    private SystemMvcMocks systemMvcMocks;

    @Test
    public void testPageMatcher200Ok() throws Exception {
        systemMvcMocks.getPageMatcher();
    }

    @Test
    public void testPageMatcherResultContains() throws Exception {
        String response = systemMvcMocks.getPageMatcher();
        assertTrue(response.contains(
            join(SEPARATOR, "show_model_grade_price_using_get", "GET:/pay/grade/model/user/UID/<userId>/show", PAGE_TYPE)));
        assertTrue(response.contains(
            join(SEPARATOR, "find_grade_payments_using_get", "GET:/pay/grade/check", PAGE_TYPE)));
    }
}

File: market/test/kotlin/ru/yandex/market/mapi/controller/PageMatcherControllerTest.kt
package ru.yandex.market.mapi.controller

import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import org.springframework.http.MediaType
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import ru.yandex.market.mapi.AbstractMapiTest
import java.nio.charset.StandardCharsets

/**
 * @author Ilya Kislitsyn / ilyakis@ / 17.01.2022
 */
class PageMatcherControllerTest : AbstractMapiTest() {
    private val SEPARATOR = "\t"
    private val PAGE_TYPE = "http"

    @Test
    fun testPageMatcherResultContains() {
        val response: String = getPageMatcher()

        assertItem(response, "main_screen_by_get", "GET:/api/screen/main")
        assertItem(response, "read_pagematch_by_get", "GET:/pagematch")
        assertItem(response, "ping", "GET:/ping")
        assertItem(response, "monitoring", "GET:/monitoring")

        // special test controller
        assertItem(response, "test_method_by_get", "GET:/abc")
        assertItem(response, "test_method_by_post", "POST:/abc")
        assertItem(response, "test_method_by_head", "HEAD:/abc")
        assertItem(response, "test_method1_by_get", "GET:/abc2")
        assertItem(response, "test_method_by_patch", "PATCH:/abc3")
    }

    private fun assertItem(response:String, operationId: String, path: String) {
        assertTrue(
            response.contains(
                listOf(operationId, path, PAGE_TYPE).joinToString(SEPARATOR)
            )
        )
    }

    private fun getPageMatcher(): String {
        return mvcCall(
            MockMvcRequestBuilders.get("/pagematch"),
            expectedType = MediaType("text", "tab-separated-values", StandardCharsets.UTF_8)
        )
    }
}